osf/metrics/reporters/institutional_users.py

```diff
@@ -0,0 +1,135 @@
+import dataclasses
+import datetime
+
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import Q, F, Sum
+
+from osf import models as osfdb
+from osf.models.spam import SpamStatus
+from addons.osfstorage.models import OsfStorageFile
+from osf.metrics.reports import InstitutionalUserReport
+from osf.metrics.utils import YearMonth
+from ._base import MonthlyReporter
+
+
+_CHUNK_SIZE = 500
+
+
+class InstitutionalUsersReporter(MonthlyReporter):
+    def report(self, yearmonth: YearMonth):
+        _before_datetime = yearmonth.next_month()
+        for _institution in osfdb.Institution.objects.all():
+            _user_qs = _institution.get_institution_users().filter(created__lt=_before_datetime)
+            for _user in _user_qs.iterator(chunk_size=_CHUNK_SIZE):
+                _helper = _InstiUserReportHelper(_institution, _user, yearmonth, _before_datetime)
+                yield _helper.report
+
+
+# helper
+@dataclasses.dataclass
+class _InstiUserReportHelper:
+    institution: osfdb.Institution
+    user: osfdb.OSFUser
+    yearmonth: YearMonth
+    before_datetime: datetime.datetime
+    report: InstitutionalUserReport = dataclasses.field(init=False)
+
+    def __post_init__(self):
+        _affiliation = self.user.get_institution_affiliation(self.institution._id)
+        self.report = InstitutionalUserReport(
+            report_yearmonth=self.yearmonth,
+            institution_id=self.institution._id,
+            user_id=self.user._id,
+            user_name=self.user.fullname,
+            department_name=(_affiliation.sso_department or None),
+            month_last_login=YearMonth.from_date(self.user.date_last_login),
+            account_creation_date=YearMonth.from_date(self.user.created),
+            orcid_id=self.user.get_verified_external_id('ORCID', verified_only=True),
```

Comment:
nit: No unverified ORCID's ? I think that's a correct choice. Good to remember in case individual users see they aren't counted and complain, more of a Product concern, but good to agree on it with Product.